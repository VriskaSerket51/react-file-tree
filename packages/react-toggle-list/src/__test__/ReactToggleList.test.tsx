import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { ReactToggleList } from '..';

describe('TEST ReactToggleList', () => {
  it('ReactToggleList Snapshot test', () => {
    const mock_list = {
      title: 'toggle',
      subItems: [
        <p onClick={() => alert('T')}>T</p>,
        <p onClick={() => alert('E')}>E</p>,
        <p onClick={() => alert('S')}>S</p>,
        <p onClick={() => alert('T')}>T</p>,
      ],
    };
    const { container } = render(<ReactToggleList items={mock_list} />);
    expect(container).toMatchSnapshot();
  });

  it('ReactToggleList Rendering', () => {
    const title = 'toggle';
    const mock_list = {
      title,
      subItems: [
        <p onClick={() => alert('T')}>T</p>,
        <p onClick={() => alert('E')}>E</p>,
      ],
    };
    render(<ReactToggleList items={mock_list} />);
    const toggleElement = screen.getByText(title);
    expect(toggleElement).toBeInTheDocument();
  });

  it('ReactToggleList init list is not showing', () => {
    const title = '토그리토글';
    const mock_list = {
      title,
      subItems: [
        <p onClick={() => alert('T')}>T</p>,
        <p onClick={() => alert('E')}>E</p>,
      ],
    };
    render(<ReactToggleList items={mock_list} />);
    expect(screen.queryByRole('list')).not.toBeInTheDocument();
  });

  it('ReactToggleList Button click event', () => {
    const title = 'toggle';
    const mock_list = {
      title,
      subItems: [
        <p onClick={() => alert('T')}>T</p>,
        <p onClick={() => alert('E')}>E</p>,
      ],
    };
    render(<ReactToggleList items={mock_list} />);
    const toggleElement = screen.getByRole('button');
    fireEvent.click(toggleElement);
    expect(screen.getByRole('list')).toBeInTheDocument();
    fireEvent.click(toggleElement);
    expect(screen.queryByRole('list')).not.toBeInTheDocument();
  });
});
