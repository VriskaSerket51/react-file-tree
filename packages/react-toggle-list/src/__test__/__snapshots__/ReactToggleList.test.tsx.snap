// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST ReactToggleList ReactToggleList Snapshot test 1`] = `
<div>
  <div>
    <div
      class="flex flex-row items-center cursor-pointer"
      role="button"
    >
      <div
        class="w-6"
      >
        <svg
          fill="none"
          height="25"
          stroke="currentColor"
          stroke-width="0"
          style="position: relative; left: -10px;"
          viewBox="0 0 15 15"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 11L6 4L10.5 7.5L6 11Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <p>
        toggle
      </p>
    </div>
  </div>
</div>
`;
